### PR TITLE
prow,manager: add support for upgrade jobs

### PR DESCRIPTION
**Pull request by AlexNPavel**
This PR adds the ability to run upgrade jobs on the modern config type.
At the time of writing this commit message, the `e2e-upgrade-all` test
type is unable to be accurately run using the step registry as the step
registry's script hardcodes the parallel conformance suite for the
`upgrade-conformance` test type, while the exisiting `e2e-upgrade-all`
chatbot test uses the full conformance suite. In this PR, the chatbot
will fall back to templates for that test type until we can reconcile
this difference.

/cc @bradmwilliams

### CODE REVIEW

File: prow.go [hunk 1]
```diff
@@ -437,6 +465,20 @@ func (m *jobManager) newJob(job *Job) error {
 			}
 			// CLUSTER_DURATION unused by tests; remove to prevent ci-operator from complaining
 			delete(matchedTarget.MultiStageTestConfiguration.Environment, "CLUSTER_DURATION")
+		}
+		if job.Mode == JobTypeUpgrade {
+			if job.Mode == JobTypeUpgrade {
```
Comment (bradmwilliams) [inline prow.go:470]: Why?

**Review by bradmwilliams — CHANGES_REQUESTED** [review]

File: prow.go (hunk 1 above)
Comment (AlexNPavel) [reply prow.go]: That was a small mistake when I was reorganizing some stuff in that section. Removed the double ifs

**Pull request merged**